addons/pos_viva_wallet/models/pos_payment_method.py

```diff
@@ -0,0 +1,207 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import json
+import base64
+import time
+import requests
+
+
+from odoo import fields, models, api, tools, _
+from odoo.exceptions import UserError, AccessError
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+TIMEOUT = 10
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    # Viva Wallet
+    viva_wallet_merchant_id = fields.Char(string="Merchant ID", help='Used when connecting to Viva Wallet: https://developer.vivawallet.com/getting-started/find-your-account-credentials/merchant-id-and-api-key/', copy=False)
+    viva_wallet_api_key = fields.Char(string="API Key", help='Used when connecting to Viva Wallet: https://developer.vivawallet.com/getting-started/find-your-account-credentials/merchant-id-and-api-key/', copy=False)
+    viva_wallet_client_id = fields.Char(string="Client ID", help='Used when connecting to Viva Wallet: https://developer.vivawallet.com/getting-started/find-your-account-credentials/pos-apis-credentials/#find-your-pos-apis-credentials', copy=False)
+    viva_wallet_client_secret = fields.Char(string="Client secret", copy=False)
+    viva_wallet_terminal_id = fields.Char(string="Terminal ID", help='[Terminal ID of the Viva Wallet terminal], for example: 16002169', copy=False)
+    viva_wallet_bearer_token = fields.Char(copy=False, groups='base.group_erp_manager', default='Bearer Token')
+    viva_wallet_webhook_verification_key = fields.Char()
+    viva_wallet_latest_response = fields.Json(copy=False, groups='base.group_erp_manager') # used to buffer the latest asynchronous notification from Adyen.
+    viva_wallet_test_mode = fields.Boolean(string="Test mode", help="Run transactions in the test environment.")
+
+    def _bearer_token(self):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
```

Comment:
```suggestion
        self.ensure_one()
        if not self.env.user.has_group('point_of_sale.group_pos_user'):
```

Seems to be used for singleton.